modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx

```diff
@@ -0,0 +1,322 @@
+	noItemsButtonLabel,
+	noItemsDescription,
+	noItemsTitle,
+	onCancelButtonClick,
+	onCreationButtonClick,
+	onOrderChange,
+	onSaveButtonClick,
+	title,
+}: OrderableTableInterface) => {
+	const [items, setItems] = useState(initialItems);
+	const [query, setQuery] = useState('');
+
+	const onSearch = (query: string) => {
+		setQuery(query);
+
+		const regexp = new RegExp(query, 'i');
+
+		setItems(
+			query
+				? initialItems.filter((item) =>
+						fields.some((field) =>
+							String(item[field.name]).match(regexp)
+						)
+				  ) || []
+				: initialItems
+		);
+	};
+
+	const handleOnOrderChange = ({
+		draggedIndex,
+		targetIndex,
+	}: {
+		draggedIndex: number;
+		targetIndex: number;
+	}) => {
+		const orderedItems = items.map((item, index) => {
+			if (index === draggedIndex) {
+				return items[targetIndex];
+			}
+
+			if (index === targetIndex) {
+				return items[draggedIndex];
+			}
+
+			return item;
+		});
+
+		setItems(orderedItems);
+
+		onOrderChange({
+			orderedItems,
+		});
+	};
+
+	return (
+		<ClayLayout.Sheet className="mt-3 orderable-table-sheet">
+			<ClayLayout.SheetHeader>
+				<h2 className="sheet-title">{title}</h2>
+			</ClayLayout.SheetHeader>
+
+			<ClayLayout.SheetSection>
+				<ManagementToolbar.Container>
+					<ManagementToolbar.ItemList expand>
+						<ManagementToolbar.Item className="nav-item-expand">
+							<Search onSearch={onSearch} query={query} />
+						</ManagementToolbar.Item>
+
+						<ManagementToolbar.Item>
+							<ClayButtonWithIcon
+								aria-label={Liferay.Language.get('add')}
+								className="nav-btn nav-btn-monospaced"
+								onClick={() => onCreationButtonClick()}
+								symbol="plus"
+							/>
+						</ManagementToolbar.Item>
+					</ManagementToolbar.ItemList>
+				</ManagementToolbar.Container>
+
+				{items.length ? (
+					<ClayTable className="orderable-table">
+						<ClayTable.Head>
+							<ClayTable.Row>
+								<ClayTable.Cell className="drag-handle-cell" />
+
+								{fields.map((field) => (
+									<ClayTable.Cell key={field.name}>
+										{field.label}
+									</ClayTable.Cell>
+								))}
+							</ClayTable.Row>
+						</ClayTable.Head>
+
+						<ClayTable.Body>
+							<DndProvider backend={HTML5Backend}>
+								{items.map((item, index) => (
+									<OrderableTableRow
+										fields={fields}
+										index={index}
+										item={item}
+										key={index}
+										onOrderChange={handleOnOrderChange}
+										query={query}
+									/>
+								))}
+							</DndProvider>
+						</ClayTable.Body>
+					</ClayTable>
+				) : query ? (
+					<ClayEmptyState
+						description={Liferay.Language.get(
+							'sorry,-no-results-were-found'
+						)}
+						title={Liferay.Language.get('no-results-found')}
+					/>
+				) : (
+					<ClayEmptyState
+						description={noItemsDescription}
+						title={noItemsTitle}
+					>
+						<ClayButton
+							displayType="secondary"
+							onClick={() => onCreationButtonClick()}
```

Comment:
Same as above, you can pass it straight through without he intermediary function delcaration